### PR TITLE
CCP & Many-to-Many Backend Changes

### PR DESCRIPTION
**Pull request by aaronabraham311**
## Overview
We completed two changes to the backend:
1. Created a `createdBy` field for collection points which is linked to a particular user
2. Defined a many-to-many relationship between resources (specifically the hospital and ambulance tables) and the events table

[Notion task](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3751aa62a2594c74b7e0ecb363304c24&p=5674871667524368a6233c64058ca4de)
## Changes
For task 1:
- Created a new migration file to add in the `createdBy` field  for collection points
- Updated schema, models, and resolver functionalities
- Defined `belongsTo` relationship between a CCP and a user

For task 2:
- Updated event, ambulances, and hospitals models to support a many-to-many relationship through new `eventAmbulances` and `eventHospitals` tables
- Updated event schema and resolver
  - updated `Event` type to have `ambulances` and `hospitals` fields
  - added `addAmbulanceToEvent` and `addHospitalToEvent` mutations (to add a single ambulance/hospital)
  - updated `addEvent` to return empty `ambulances` and `hospitals` arrays
  - updated `updateEvent` (to add an arbitrary number of ambulances and hospitals)
- Updated ambulance and hospital deletion mutations to delete associations with events

## Screenshots
**Screenshot of task 1:**
Screen Shot 2020-06-01 at 8 43 41 PM

- Manual GraphQL testing
**Screenshots of testing for new many-to-many relationship between resources and events**

_query:_
image

_addEvent:_
addEvent

_updateEvent:_
updateEvent

_updateEvent with invalid hospital ID:_
updateEvent invalid hospital ID

_updateEvent with invalid ambulance ID:_
updateEvent invalid ambulance ID

_addHospitalToEvent:_
addHospitalToEvent

_addHospitalToEvent already assigned:_
image

_addHospitalToEvent invalid event ID:_
addHospitalToEvent invalid event ID

_addHospitalToEvent invalid hospital ID:_
addHospitalToEvent invalid hospital ID

_addAmbulanceToEvent:_
addAmbulanceToEvent

_addAmbulanceToEvent already assigned:_
addAmbulanceToEvent already assigned

_addAmbulanceToEvent invalid ambulance ID:_
addAmbulanceToEvent invalid ambulance ID

_addAmbulanceToEvent invalid event ID:_
addAmbulanceToEvent invalid event ID

## Checklist
- [x] Change title to be more descriptive (shouldn't be just branch name)
- [x] Update `Notion task` link
- [x] `Overview`
- [x] `Changes`
  - [ ] \(Optional) Add inline comments in diff
- [x] \(Optional) `Screenshots`
- [x] \(Optional) `Testing` (N/A if nothing changed)
- [ ] \(Optional) `Other Notes`
- [ ] Request at least 2 reviewers
- [x] Assign yourself to the PR
- [x] Make sure build passes

@uwblueprint/paramedics

### CODE REVIEW

File: resolvers/event.js [hunk 1]
```diff
@@ -17,41 +36,230 @@ const eventResolvers = {
...
+        {
+          model: db.hospital,
+          attributes: ['id', 'name', 'createdAt', 'updatedAt']
+        }]
+      });   
+    },
+    deleteEvent: async (parent, args) => {
       // Return status for destroy
       // 1 for successful deletion, 0 otherwise
+      await db.eventAmbulances.destroy({
```
Comment (danielchen115) [inline resolvers/event.js:251]: Wonder if we can set up delete cascades to deal foreign key deletions, but this is good for now!

File: resolvers/event.js [hunk 2]
```diff
@@ -17,41 +36,230 @@ const eventResolvers = {
...
+    addHospitalToEvent: async (parent, args) => {
+      // Checking if event exists
+      const event = await db.event.findByPk(args.eventId);
+      if (!event) {
+        throw new Error("Invalid event ID");
+      }
+      
+      // Checking if hospital exists
+      const hospital = await db.hospital.findByPk(args.hospitalId['id']);
+      if (!event) {
```
Comment (carelynntsai) [inline resolvers/event.js:215]: Shouldn't this be `if (!hospital) {`

Comment (carelynntsai) [inline resolvers/event.js:215]:
```suggestion
      if (!hospital) {
```

Comment (carelynntsai) [reply resolvers/event.js:215]: I believe due to this small error I can add non-existent hospitals
image
image
Other than that looks good! Great job guys :)

**Pull request merged**